app/models/shop.rb

```diff
@@ -2,6 +2,10 @@ class Shop < ActiveRecord::Base
   belongs_to :user
   has_many :products, :dependent => :delete_all
 
+  validates :name, length: {minimum: 2, maximum: 15}, presence:true
+  validates :description, length: {minimum: 2}, presence:true
+  validates :email_id, presence:true
```

Comment:
@danielpaul It does verify. The reason why I validated email_id was because what if a customer wanted to contact the shop owner?